File: src/frontend/src/signal/sampleScaling.ts
/**
 * Escalado del entero recibido por el puerto serie a mV (feature 005, FR-006).
 * Span = (10-0)/(4095-0), Zero = 0.
 */
const SPAN = (10 - 0) / (4095 - 0);
const ZERO = 0;

export function toMillivolts(count: number): number {
  return count * SPAN + ZERO;
}

File: src/backend/ECGViewer.Api/Serial/SampleScaling.cs
namespace ECGViewer.Api.Serial;

/// <summary>
/// Escalado de una cuenta entera recibida por el puerto serie a milivoltios
/// (feature 005, FR-006). Span/Zero son los valores fijos del cálculo provisto:
/// Span = (10 − 0) / (4095 − 0) ≈ 0.00244 mV/cuenta; Zero = 0.
/// </summary>
public static class SampleScaling
{
    private const double Span = (10.0 - 0.0) / (4095.0 - 0.0);
    private const double Zero = 0.0;

    public static double ToMillivolts(int count) => count * Span + Zero;
}
